ERCS/erc-7604.md

```diff
@@ -0,0 +1,202 @@
+---
+eip: 7604
+title: ERC-1155 Permit Approvals
+description: Permit approvals for ERC-1155 tokens
+author: calvbore (@calvbore)
+discussions-to: https://ethereum-magicians.org/t/proposal-for-a-new-eip-erc-2612-style-permits-for-erc1155-nfts/15504
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-01-27
+requires: 165, 712, 1155, 5216
+---
+
+## Abstract
+
+The "permit" approval flow for both [ERC-20](./erc-20.md) and [ERC-721](./erc-721.md) are large improvements for the existing UX of the token underlying each ERC. This ERC extends the "permit" pattern to [ERC-1155](./erc-20.md) tokens, borrowing heavily upon both [ERC-4494](./erc-4494.md) and [ERC-2612](./erc-2612.md).
+
+The structure of [ERC-1155](./erc-1155.md) tokens requires a new ERC to account for the token standard's use of both token IDs and balances (also why this ERC requires [ERC-5216](./erc-5216.md)). An additional field of arbitrary `bytes` is added for extra data to be added to the signatures in the case that the permit function would be extended in some way.  
+
+## Motivation
+
+The permit structures outlined in both [ERC-4494](./erc-4494) and [ERC-2612](./erc-2612) allows a signed message to create an approval, but are only applicable to their respective underlying tokens ([ERC-721](./erc-721) and [ERC-20](./erc-20)).
+
+While the permit flow described in the above ERCs greatly improves UX and contract interactions they do not allow for any extensibility to the permit flow. It may be that implementors would like to verify additional data than what is strictly required by the permit signature.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+Three new functions must be added to ERC-1155 and ERC-5216.
+
+```solidity
+interface IERC1155Permit {
+	function permit(address owner, address operator, uint256 tokenId, uint256 value, uint256 deadline, bytes memory data, bytes memory sig) external;
+	function nonces(address owner, uint256 tokenId) external view returns (uint256);
+	function DOMAIN_SEPARATOR() external view returns (bytes32);
+}
+```
+
+The semantics of which are as follows:
+
+For all addresses `owner`, `spender`, uint256's `tokenId`, `value`, `deadline`, and `nonce`, bytes `data` and `sig`, a call to `permit(owner, spender, tokenId, value, deadline, data, sig)` MUST set `allownace(owner, spender, tokenId)` to `value`, increment `nonces(owner, tokenId)` by 1, and emit a corresponding `ApprovalByAmount` event, if and only if the following conditions are met:
```

Comment:
There's a typo here `allownace` should be `allowance`

Reply:
Thanks! updated!